tests/unit/test_charm_2.py

```diff
@@ -1,12 +1,48 @@
+import os
 import unittest
 import unittest.mock
 
 import ops.testing
 from charm import WordpressCharm
 
+METADATA_FILE = os.path.join(os.path.split(__file__)[0], "..", "..", "metadata.yaml")
+
 
 class TestWordpressK8s(unittest.TestCase):
     def setUp(self):
+        container_fs = {}
+        self.container_fs = container_fs
+        installed_database_hosts = set()
+        self.installed_database_hosts = installed_database_hosts
+
+        def mock_current_wp_config(_self):
+            container_fs.get(_self._wp_config_path())
+
+        def mock_push_wp_config(_self, wp_config):
+            container_fs[_self._wp_config_path()] = wp_config
+
+        def get_current_db_host(_self):
+            db_host = _self.model.config["db_host"]
+            if not db_host:
+                db_host = _self.state.db_host
+            return db_host
+
+        def mock_wp_is_installed(_self):
+            db_host = get_current_db_host(_self)
+            return db_host in installed_database_hosts
+
+        def mock_wp_install(_self):
+            db_host = get_current_db_host(_self)
+            installed_database_hosts.add(db_host)
+
+        self.container_patch = unittest.mock.patch.multiple(
+            WordpressCharm,
```

Comment:
Any particular reason for not using the https://docs.python.org/3/library/unittest.mock.html library?

Reply:
I think this is `unitest.mock`.

Reply:
Yes, I was more meaning, for example, using a mock.MagicMock instead of defining a functions like `mock_wp_is_installed`, although I see a bit of custom logic that might be difficult to implement using a MagicMock

Reply:
Yes, this is a simulation of container file system and process execution, so it's more complex than MagicMock

Reply:
👍